lib/build/cpp/cxx

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env sh
+set -e
+
+# There is something more mysterious going on here wrt crt* objects
+# The compiler seems to be finding the ones in sysroot/lib and adding them
+# to the linker. So will it also find crtbegin from compiler-rt if we tell it to
+# look there, or would we need to rebuild clang with USE_COMPILER_RT enabled?
```

Comment:
So I think we can probably do something better re these crt{begin/end} files. For starters we could just relocate the compiler-rt ones and overwrite the dummy placeholders we already have. Or possibly we could rebuild clang with `CLANG_DEFAULT_RTLIB=compiler-rt` which I believe configures it use compiler-rt generated bits instead. 

On a related note there is also `CLANG_DEFAULT_CXX_STDLIB` which if we set to `libc++` I think would then bake that flag in to the compiler. These two points, plus a few other things make me wonder whether we should do a fresh build of clang once we've bootstrapped our own sysroot with our musl libc and built libc++, compiler-rt etc so that we then have a version of clang that is more aware of our custom built environment and it would also be more consistent across environments. Probably one to ponder for a bit and to implement another day when we've got some stability around what we exactly need in the toolchain for c++.